plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -546,8 +547,12 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                   if( !exhausted )
                      exhausted = block_is_exhausted();
                } else {
-                  auto e_ptr = trace->except->dynamic_copy_exception();
-                  send_response( e_ptr );
+                  if( return_failure_trace ) {
+                     send_response( trace );
```

Comment:
This will make the log messages from `send_response` wrong.

Reply:
Good point. Updated.